fastchat/serve/monitor/clean_chat_data.py

```diff
@@ -44,90 +56,125 @@ def get_log_files(max_num_files=None):
-                raw_data.append(row)
+def get_action_type_data(filename, action_type):
+    for _ in range(5):
+        try:
+            lines = open(filename).readlines()
+            break
+        except FileNotFoundError:
+            time.sleep(2)
 
-    all_models = set()
-    all_ips = dict()
-    chats = []
+    for l in lines:
+        row = json.loads(l)
+        if row["type"] == action_type:
+            return row
+
+
+def process_data(row, action_type, all_ips):
+    # Initialize local counters
     ct_invalid_conv_id = 0
     ct_invalid = 0
     ct_network_error = 0
-    for row in raw_data:
-        try:
-            if action_type in ["chat", "upvote", "downvote"]:
-                state = row["state"]
-                model = row["model"]
-            elif action_type == "leftvote":
-                state = row["states"][0]
-                model = row["states"][0]["model_name"]
-            elif action_type == "rightvote":
-                state = row["states"][1]
-                model = row["states"][1]["model_name"]
-            conversation_id = state["conv_id"]
-        except KeyError:
-            ct_invalid_conv_id += 1
-            continue
 
-        if conversation_id is None:
-            ct_invalid_conv_id += 1
-            continue
+    try:
+        if action_type in ["chat", "upvote", "downvote"]:
+            state = row["state"]
+            model = row["model"]
+        elif action_type == "leftvote":
+            state = row["states"][0]
+            model = row["states"][0]["model_name"]
+        elif action_type == "rightvote":
+            state = row["states"][1]
+            model = row["states"][1]["model_name"]
+        conversation_id = state["conv_id"]
+    except KeyError:
+        ct_invalid_conv_id += 1
+        return None, ct_invalid_conv_id, ct_invalid, ct_network_error, None
+
+    if conversation_id is None:
+        ct_invalid_conv_id += 1
+        return None, ct_invalid_conv_id, ct_invalid, ct_network_error, None
+
+    conversation = to_openai_format(state["messages"][state["offset"] :])
+    if not isinstance(model, str):
+        ct_invalid += 1
+        return None, ct_invalid_conv_id, ct_invalid, ct_network_error, None
+    model = replace_model_name(model, row["tstamp"])
+
+    try:
+        lang_code = detect_language(state["messages"][state["offset"]][1])
+    except IndexError:
+        ct_invalid += 1
+        return None, ct_invalid_conv_id, ct_invalid, ct_network_error, None
+
+    if not all(isinstance(x["content"], str) for x in conversation):
+        ct_invalid += 1
+        return None, ct_invalid_conv_id, ct_invalid, ct_network_error, None
+
+    messages = "".join([x["content"] for x in conversation]).lower()
+    if NETWORK_ERROR_MSG in messages:
+        ct_network_error += 1
+        return None, ct_invalid_conv_id, ct_invalid, ct_network_error, None
+
+    ip = row["ip"]
+    # Synchronize access to all_ips using the lock
+    with LOCK:
```

Comment:
This might slow us down significantly. why don't we just change this ip -> user_id with a hash function? instead of counting which seems unnecessary.

Reply:
Makes sense. How about `uuid.uuid4().hex`?

Reply:
We want to a hash function, so that same ip is mapped to same id

Reply:
Oh mb, I meant something like `hashlib.md5(ip_address.encode())`.